tsconfig.base.json

```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "module": "ES2022",
+    "module": "node16",
```

Comment:
This is a scary config change! Are you confident that it isn't going to break bundling, etc, for kmc, developer/server, etc?

Reply:
Yes.  This is the prescribed "correct" way to do things in TS 5; we _cannot_ leave `ES2022` in place, as it's now a TS config error to do so.  Use of `"moduleResolution": "node16"` explicitly requires use of `"module": "node16"`.  They _must_ match.  We cannot update to TS 5.0+ without doing this.

Would you like me to add a user test that verifies that the resulting Developer build is unaffected?  If so, would you prefer it here, on the final related PR (#11464), or both?

Reply:
Okay, that makes sense, thanks.

It'd be good for the user test to run on the final PR in the chain. It _should_ be fine, as far as I can tell, but I am getting increasingly wary with config changes and unanticipated outcomes in js/ts world, particularly where we have so many modules with varying histories.

I think we need user tests of the compile toolchain (compiling within Keyman Developer IDE is fine) and Keyman Developer Server (so testing web keyboards in Keyman Developer IDE).

Reply:
FWIW, the Developer build from this PR (which passed) seems to have a fully-functioning Web test-host page when I test it locally myself.  Recompilation of the keyboard (after a .keyman-touch-layout edit) seems to proceed fine.

That said... I noticed something really strange.  If I edited the touch-layout a bit, then built the keyboard and tested via Developer Server, all was fine.  If I then returned to the touch-layout editor and used `CTRL-Z` (undo) once, saved, then rebuilt the keyboard and resumed testing... the results did _not_ get updated.  I was able to determine that the touch-layout (undo) edits did not make it into the compiled .js in this case.

Re-editing the touch layout, then copying-and-pasting the original value for edited parts of the touch layout back in place, then doing the usual save -> rebuild -> test, acted as expected, updating the test-host keyboard properly once again.

Reply:
> Looks to be something undo-specific?

Possibly yes.

Reply:
Forgot to link it previously:  I spun that "something really strange" observation off into #11514.